src/pages/project/helpers/projectRoleHelpers.ts

```diff
@@ -60,29 +70,181 @@ export const checkIfSameAffiliationOnSameRoleType = (
   return role.affiliation?.type === 'Organization' && role.affiliation?.id === newAffiliationId && isSameRoleType;
 };
 
-const findProjectManagerRoleThatHasAffiliation = (contributor: ProjectContributor, affiliationId: string) => {
-  return contributor.roles.findIndex((role) => role.affiliation?.id === affiliationId && isProjectManagerRole(role));
+const findProjectManagerRoleThatHasAffiliation = (roles: ProjectContributorRole[], affiliationId: string) => {
+  return roles.findIndex((role) => role.affiliation?.id === affiliationId && isProjectManagerRole(role));
 };
 
-const findNonProjectManagerRoleThatHasAffiliation = (contributor: ProjectContributor, affiliationId: string) => {
-  return contributor.roles.findIndex((role) => role.affiliation?.id === affiliationId && isNonProjectManagerRole(role));
+const findNonProjectManagerRoleThatHasAffiliation = (roles: ProjectContributorRole[], affiliationId: string) => {
+  return roles.findIndex((role) => role.affiliation?.id === affiliationId && isNonProjectManagerRole(role));
 };
 
 export const findRoleIndexForAffiliation = (
   asProjectManager: boolean,
-  contributor: ProjectContributor,
+  roles: ProjectContributorRole[],
   affiliationId: string
 ) => {
   return asProjectManager
-    ? findProjectManagerRoleThatHasAffiliation(contributor, affiliationId)
-    : findNonProjectManagerRoleThatHasAffiliation(contributor, affiliationId);
+    ? findProjectManagerRoleThatHasAffiliation(roles, affiliationId)
+    : findNonProjectManagerRoleThatHasAffiliation(roles, affiliationId);
 };
 
 export const notLastOfItsRoleType = (
-  contributor: ProjectContributor,
+  roles: ProjectContributorRole[],
   affiliationId: string,
   roleType: ProjectContributorType
 ) => {
-  const rolesWithoutAffiliationId = contributor.roles.filter((role) => role.affiliation?.id !== affiliationId);
+  const rolesWithoutAffiliationId = roles.filter((role) => role.affiliation?.id !== affiliationId);
   return rolesWithoutAffiliationId.some((role) => role.type === roleType);
 };
+
+export enum AffiliationErrors {
+  NO_AFFILIATION_ID,
+  CAN_ONLY_BE_ONE_PROJECT_MANAGER,
+  ADD_DUPLICATE_AFFILIATION,
+  NO_ROLE_TO_CHANGE,
+  NO_ROLE_TO_REMOVE,
+}
+
+export const addAffiliation = (
+  newAffiliationId: string,
+  contributorRoles: ProjectContributorRole[],
+  asProjectManager = false,
+  labels?: LanguageString
```

Comment:
snakka vi ikke om å droppe `labels` helt for contributors på samme måte som i resultat-wizard, eller er det noen tilfeller hvor vi ønsker det også?

Reply:
Jeg har gått over til at det er mulig å ikke legge til labels, men det er ikke en ulempe om det legges til, tenker jeg.